mart/callbacks/adversary_connector.py

```diff
@@ -66,8 +71,27 @@ def on_after_batch_transfer(self, pl_module, batch, dataloader_idx):
         # Move adversary to same device as pl_module and run attack
         adversary.to(pl_module.device)
 
-        # Directly pass batch instead of assuming it has a structure.
-        adversary.fit(batch=batch, model=pl_module)
-        batch_adv = adversary(batch=batch)
+        # Make a simple model interface that outputs=model(input, target)
+        def model(input, target):
+            batch = self.batch_c15n.revert(input, target)
+
+            if hasattr(pl_module, "attack_step"):
+                outputs = pl_module.attack_step(batch, dataloader_idx)
+            elif hasattr(pl_module, "training_step"):
+                # Disable logging if we have to reuse training_step() of the target model.
+                with MonkeyPatch(pl_module, "log", lambda *args, **kwargs: None):
+                    outputs = pl_module.training_step(batch, dataloader_idx)
+            else:
+                outputs = model(batch)
```

Comment:
Is this right? Shouldn't it be pl_module? You'll probably want to write a test for each of these cases.

Reply:
Good catch. Fixed in 16443c2b54ebae655c0d25332ddc3fc67e3afe79

Reply:
I deleted the last line, because `training_step` is always defined in `LightningModule`. dbf8b9e3bf0615db585bf14abee018b74442e95e

Reply:
Added test in 3df41b381ef250ba6a3e00190a0600ced2fcc892